examples/StoreApi/Program.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Momento.Sdk.StoreClient;
+
+namespace Driver;
+
+public class Program
+{
+    public static async Task Main(string[] args)
+    {
+        var client = new StoreClient();
+        await client.CreateStoreAsync("my-store");
+
+        // Can use implicit cast to create StoreValue.
+        await client.PutAsync("my-store", "my-key", "my-value");
+        await client.PutAsync("my-store", "my-key", 42);
+        await client.PutAsync("my-store", "my-key", 3.14);
+        await client.PutAsync("my-store", "my-key", true);
+
+        // The compiler will detect invalid casts.
+        // await client.PutAsync("my-store", "my-key", new string[] { "a", "b" });
+
+        // This demos usage where the developer is sure of the type.
+        {
+            var response = await client.GetAsync("my-store", "string-value");
+            if (response is StoreGet.Success success)

```

Comment:
have you seen this before?

https://spencerfarley.com/2021/03/26/unions-in-csharp/#improvements-in-c

Reply:
It rings a bell. I think C# 9 wasn't out at the time and was a preview.